### ISSUE TITLE
Error in dimnames(slot(x, "coords")) <- `*vtmp*` :    length of 'dimnames' [1] not equal to array extent

### ISSUE THREAD

**Issue by bniebuhr** [16EAGLE/moveVis#78]
Hello!

I am having an issue for animating movement for some animals. The code works just fine for animals monitored in some years, but bad in some other years. The problem seem to be specifically at the `align_move` function - and, from my debug exercises, specifically in these commands within the function:

```R
m <- moveStack(lapply(m.indi, function(x) {
        ts.m <- timestamps(x)
        ts.t <- ts.target[ts.target >= min(ts.m) & ts.target <= 
            max(ts.m)]
        interpolateTime(x, ts.t, spaceMethod)
    }))
```

Below I show my code, I can also share some data by email if necessary for a reproducible example.

```R
library(move)
library(moveVis)
library(amt)

# read data after some cleaning
load("files_loaded.RData")

# get data for only one year
# it works for most years, but not for some
yy <- 2014
winter.years <- winter.tracks %>% 
  dplyr::filter(year == yy, !is.na(x_) | !is.na(y_)) %>% 
  dplyr::mutate(id = as.factor(as.character(id))) %>% 
  amt::mk_track(.x = x_, .y = y_, .t = t_, id = id, crs = CRS("+init=epsg:32633")) %>% 
  amt::transform_coords(crs_to = CRS("+init=epsg:4326")) %>% 
  dplyr::arrange(id, t_)

# check
winter.years$id

# transform it into move object
trk.object <- get("winter.year") # this is to prepare the code for a function or loop later, for all years
winter.year.move <- move(x = trk.object$x_, y = trk.object$y_, 
                         time = trk.object$t_,
                         proj = CRS("+init=epsg:4326"), 
                         animal = trk.object$id, sensor = "GPS")

# animate

# align move_data to a uniform time scale
### HERE I GET THE ERROR
m <- align_move(winter.2009.move, res = 1, digit = 0, unit = "days")

# create spatial frames with a OpenStreetMap watercolour map
limits <- sf::st_bbox(winter.area)[c(1,3,2,4)]

frames <- frames_spatial(m, path_colours = rainbow(n = length(levels(m@trackId))),
                         map_service = "osm", map_type = "watercolor", alpha = 0.5,
                         ext = raster::extent(limits)) %>% 
  add_gg(gg = expr(geom_sf(data = windfarms, shape = 4))) %>% 
  add_gg(gg = expr(geom_sf(data = buffer.wf, fill = NA))) %>% 
  add_gg(gg = expr(geom_sf(data = powerline, col = "red"))) %>% 
  add_gg(gg = expr(geom_sf(data = winter.range, fill = NA, linetype = "dashed"))) %>% 
  add_gg(gg = expr(theme(legend.position = "none"))) %>% 
  #add_gg(gg = expr(lims(x = limits[1:2], y = limits[3:4]))) %>% 
  add_labels(x = "Longitude", y = "Latitude") %>% # add some customizations, such as axis labels
  add_northarrow() %>% 
  add_scalebar(distance = 10, position = "upperleft") %>% 
  add_timestamps(m, type = "label") %>% 
  add_progress()

frames.box <- frames %>%  # preview one of the frames, e.g. the 100th frame
  moveVis::add_gg(gg = expr(lims(x = limits[1:2], y = limits[3:4])))
    
# animate frames
animate_frames(frames.box, out_file = "animation_2014.gif", fps = 16, overwrite = T)
```

The complete error I get during the function `align_move` is:
```
Error in dimnames(slot(x, "coords")) <- `*vtmp*` : 
  length of 'dimnames' [1] not equal to array extent
In addition: Warning messages:
1: In max.default(numeric(0), na.rm = FALSE) :
  no non-missing arguments to max; returning -Inf
2: In min.default(numeric(0), na.rm = FALSE) :
  no non-missing arguments to min; returning Inf
3: In max.default(numeric(0), na.rm = FALSE) :
  no non-missing arguments to max; returning -Inf
4: In min.default(numeric(0), na.rm = FALSE) :
  no non-missing arguments to min; returning Inf
```

Do anyone have an idea of what may be the problem?
Thanks in advance!

**Comment by bniebuhr** [comment]
It seems other people have had the same problem already: https://stackoverflow.com/questions/55777725/error-in-dimnamesslotx-coords-when-running-align-move-in-move-package

**Comment by bniebuhr** [comment]
Hello!

For the hints from this [other person who had a similar problem](https://stackoverflow.com/questions/55777725/error-in-dimnamesslotx-coords-when-running-align-move-in-move-package), I removed the animals with less than 10 locations for the dataset, and this error in the `align_move` function did not appear. The code worked fine. So the issue is solved.

It would be nice to know, however, why this happened, as well as the minimum amount of locations necessary to run this function. Maybe this could be in the documentation of the function, and also raise a warning or error message in case there are some individuals with less than 10 (or the minimum number of points) necessary to run it.

**Comment by bniebuhr** [comment]
Just as a remark: I tried now with individuals with 30 locations, and it also did not work.

**Comment by 16EAGLE** [comment]
Hi, sorry for my late reply. I would like to resolve this issue – would it be possible that you send me a subset of your data that creates the error and the align settings you use? Thanks for reporting this

**Comment by bniebuhr** [comment]
Yes, thanks for that! In which email should I send it?

**Comment by 16EAGLE** [comment]
jakob.schwalb-willmann[at]uni-wuerzburg.de

**Comment by 16EAGLE** [comment]
have you already sent it? not sure if I received something

**Comment by martinpmBD** [comment]
I got the same issue. I had a device with only 6 locations. Once I removed it, this error with align_move disappeared. The other devices had > 80 locations.

**Comment by 16EAGLE** [comment]
@martinpmBD Could you provide a reproducible example (e.g. along with those 6 locations) which results in the error? This would make it easier for me  to trace the cause of the issue.

**Comment by Arcticweasel** [comment]
I got the same issue. I have only observations with 2 locations. It works for some subsets of my data and for some not. I attached 2 subsets of my data. The first works, the second doesn't. Here is the code I used.

```
library(readxl)
library(move)
library(moveVis)

working <- read_excel("working_data.xlsx")
geo_move <- df2move(working, proj = "+proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0", 
                         track_id = "track_id", x = "station_long", y = "station_lat", time = "timestamp" )
geo_aligned <- align_move(geo_move, digit = 0, unit = "hours")

not_working <- read_excel("not_working_data.xlsx")
geo_move <- df2move(not_working, proj = "+proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0", 
                         track_id = "track_id", x = "station_long", y = "station_lat", time = "timestamp" )
geo_aligned <- align_move(geo_move, digit = 0, unit = "hours")
```

Thank you in advance!


[working_data.xlsx](https://github.com/16EAGLE/moveVis/files/4515971/working_data.xlsx)
[not_working_data.xlsx](https://github.com/16EAGLE/moveVis/files/4515974/not_working_data.xlsx)

**Comment by 16EAGLE** [comment]
Thanks for the data and code. I've eventually found the time to go to the bottom of this issue. It should now be resolved (f9eb5c55ecdcf88f1ef0198dd1635c6daceb8b88):

* `align_move()` now warns the user when a temporal resolution is selected that cannot be applied to at least one trajectory and excludes the trajectory/trajectories in question from the returned object or fails with a specific error if that is the case for all trajectories. In such cases, you may want to define a finder resolution using `res` and the respective unit using `unit`. As before, `res` defaults to the minimum temporal resolution found in `m`.
* Argument `digit` is now deprecated, since digits are now calculated automatically. Existing code using `digit` will still work.
* Small improvements, including printing the resolution to which `m` will be aligned.

The changes are also documented here: http://movevis.org/news/index.html. Please also check the updated help of `align_move()`: http://movevis.org/reference/align_move.html

Please reinstall `moveVis` from GitHub using `remotes::install_github("16eagle/moveVis")`. 

Here a reprex with the example data:

``` r
library(moveVis)
data("move_data", package = "moveVis")

m <- align_move(move_data) # default: minimum resolution
#> Temporal resolution of 3.96 [mins] is used to align trajectories.

m <- align_move(move_data, res = 4, unit = "mins")
#> Temporal resolution of 4 [mins] is used to align trajectories.

# attempting to align move_data to a resolution greater than the full
# temporal coverage of some trajectories:
m <- align_move(move_data, res = 13, unit = "hours")
#> Temporal resolution of 13 [hours] is used to align trajectories.
#> Warning: The full temporal coverage of at least one trajectory is shorter than the specified resolution. You may want to choose a finer resolution.
#> Trajectories that are excluded: 'T342g', 'T932u'
```

<sup>Created on 2020-04-30 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Here a reprex with the data subset by @Arcticweasel that previously caused an error by `align_move()`:

``` r
library(readxl)
library(move)
library(moveVis)

# previously not working
not_working <- read_excel("not_working_data.xlsx")
geo_move <- df2move(not_working, proj = "+proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0", 
                    track_id = "track_id", x = "station_long", y = "station_lat", time = "timestamp" )

geo_aligned <- align_move(geo_move) # default: minimum resolution
#> Temporal resolution of 1.53 [mins] is used to align trajectories.

# using a resolution greater than the full temporal coverage of some trajectories
geo_aligned <- align_move(geo_move, res = 5, unit = "mins")
#> Temporal resolution of 5 [mins] is used to align trajectories.
#> Warning: The full temporal coverage of at least one trajectory is shorter than the specified resolution. You may want to choose a finer resolution.
#> Trajectories that are excluded: 'X2', 'X5', 'X37', 'X38', 'X46', 'X78', 'X83', 'X119', 'X159', 'X182', 'X184', 'X187', 'X230'
```

<sup>Created on 2020-04-30 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Could you report me if this fix solves the issue for you? Thanks for your patience.

**Comment by Arcticweasel** [comment]
Thank you very much! This solved my issue completely.
